src/NSubstitute/Exceptions/AmbiguousArgumentsException.cs

```diff
@@ -6,6 +6,12 @@ namespace NSubstitute.Exceptions
 {
     public class AmbiguousArgumentsException : SubstituteException
     {
+        private static readonly string BaseMessage =
```

Comment:
Do you see any value in splitting the message constant and the actual string with arguments? I thought that string interpolation make it look better, so the variable and static parts are bundled seamlessly and you see the result at once, without switching between constant and concatenation..

The same is valid for the message below.

Reply:
As the message got longer I thought it was nice to separate out the dynamic and the static part.  This is subjective so I don't mind either way. :) 

Does the string interpolation stuff get compiled down to a constant where possible (the parts like `$"Some text{Environment.NewLine}" +$"More text{Environment.NewLine}"`)? Or is it dynamically created each time? If dynamic it might be better to keep it separate so it only creates the constant part of the string once.

Reply:
>  Or is it dynamically created each time?

I've just checked. In this particular case it performs a concatenation of the multiple strings, formatted using the `string.Format()` call. However, I don't see any reason to worry about that. This is not a hot place, moreover - it's an exceptional case. Therefore, it doesn't matter how much strings we need to concatenate to raise an exception - those things are supposed to never happen 😅

In this case I'd prefer to focus on the code readability and, personally, I feel that interpolation is read better.

Let's probably also see @alexandrnikitin opinion, so it's clear for me what to do 😉

Reply:
Either way works for me. I would got with string interpolation. You can go with multi line string interpolation using `$@" ...  "` It will use only one call to `string.Format` but I wouldn't worry about allocations here.